shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -249,17 +181,8 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
     return NO;
   }
 
-  const FlutterRendererConfig rendererConfig = {
-      .type = kOpenGL,
-      .open_gl.struct_size = sizeof(FlutterOpenGLRendererConfig),
-      .open_gl.make_current = (BoolCallback)OnMakeCurrent,
-      .open_gl.clear_current = (BoolCallback)OnClearCurrent,
-      .open_gl.present = (BoolCallback)OnPresent,
-      .open_gl.fbo_with_frame_info_callback = (UIntFrameInfoCallback)OnFBO,
-      .open_gl.fbo_reset_after_present = true,
-      .open_gl.make_resource_current = (BoolCallback)OnMakeResourceCurrent,
-      .open_gl.gl_external_texture_frame_callback = (TextureFrameCallback)OnAcquireExternalTexture,
-  };
+  [_openGLRenderer attachToFlutterView:_viewController.flutterView];
```

Comment:
`attachToFlutterView:`'s parameter is marked non-null, but there's no guarantee that there's a `_viewController`. You need the same logic you have in `setViewController:` here as well. Alternately, you could make this kind of mistake impossible and remove duplicated code by having `attachToFlutterView:` allow nil (maybe changing its name to `setFlutterView:`), and remove `clearResourceContext`.